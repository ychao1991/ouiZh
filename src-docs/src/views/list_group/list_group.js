/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React, { useState, Fragment } from 'react';

import {
  OuiListGroup,
  OuiListGroupItem,
  OuiSpacer,
  OuiSwitch,
  OuiCode,
  OuiFlexGroup,
  OuiFlexItem,
} from '../../../../src/components';

export default () => {
  const [flushWidth, setFlushWidth] = useState(false);
  const [showBorder, setShowBorder] = useState(false);

  return (
    <Fragment>
      <OuiFlexGroup alignItems="center">
        <OuiFlexItem grow={false}>
          <OuiSwitch
            label={
              <span>
                Show as <OuiCode>flush</OuiCode>
              </span>
            }
            checked={flushWidth}
            onChange={() => setFlushWidth(!flushWidth)}
          />
        </OuiFlexItem>
        <OuiFlexItem grow={false}>
          <OuiSwitch
            label={
              <span>
                Show as <OuiCode>bordered</OuiCode>
              </span>
            }
            checked={showBorder}
            onChange={() => {
              setShowBorder(!showBorder);
            }}
          />
        </OuiFlexItem>
      </OuiFlexGroup>

      <OuiSpacer size="l" />

      <OuiListGroup flush={flushWidth} bordered={showBorder}>
        <OuiListGroupItem onClick={() => {}} label="First item" />

        <OuiListGroupItem onClick={() => {}} label="Second item" />

        <OuiListGroupItem onClick={() => {}} label="Third item" isActive />

        <OuiListGroupItem onClick={() => {}} label="Fourth item" isDisabled />
      </OuiListGroup>
    </Fragment>
  );
};
