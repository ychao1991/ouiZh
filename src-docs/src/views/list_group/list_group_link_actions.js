/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React, { useState } from 'react';

import { OuiListGroup, OuiListGroupItem } from '../../../../src/components';

export default () => {
  const [favorite1, setFavorite1] = useState(undefined);
  const [favorite2, setFavorite2] = useState('link2');
  const [favorite3, setFavorite3] = useState(undefined);
  const [favorite4, setFavorite4] = useState(undefined);

  const link1Clicked = () => {
    setFavorite1(favorite1 === 'link1' ? undefined : 'link1');
    if (favorite1 === undefined) {
      document.activeElement.blur();
    }
  };

  const link2Clicked = () => {
    setFavorite2(favorite2 === 'link2' ? undefined : 'link2');
    if (favorite2 === undefined) {
      document.activeElement.blur();
    }
  };

  const link3Clicked = () => {
    setFavorite3(favorite3 === 'link3' ? undefined : 'link3');
    if (favorite3 === undefined) {
      document.activeElement.blur();
    }
  };

  const link4Clicked = () => {
    setFavorite4(favorite4 === 'link4' ? undefined : 'link4');
    if (favorite3 === undefined) {
      document.activeElement.blur();
    }
  };

  return (
    <OuiListGroup maxWidth={288}>
      <OuiListGroupItem
        id="link1"
        iconType="bullseye"
        label="OUI button link"
        onClick={() => {}}
        isActive
        extraAction={{
          color: 'subdued',
          onClick: link1Clicked,
          iconType: favorite1 === 'link1' ? 'starFilled' : 'starEmpty',
          iconSize: 's',
          'aria-label': 'Favorite link1',
          alwaysShow: favorite1 === 'link1',
        }}
      />

      <OuiListGroupItem
        id="link2"
        iconType="visualizeApp"
        onClick={() => {}}
        label="OUI button link"
        extraAction={{
          color: 'subdued',
          onClick: link2Clicked,
          iconType: favorite2 === 'link2' ? 'starFilled' : 'starEmpty',
          iconSize: 's',
          'aria-label': 'Favorite link2',
          alwaysShow: favorite2 === 'link2',
        }}
      />

      <OuiListGroupItem
        id="link3"
        iconType="lensApp"
        iconProps={{ color: 'default' }}
        onClick={() => {}}
        label="OUI button link"
        extraAction={{
          color: 'subdued',
          onClick: link3Clicked,
          iconType: favorite3 === 'link3' ? 'starFilled' : 'starEmpty',
          iconSize: 's',
          'aria-label': 'Favorite link3',
          alwaysShow: favorite3 === 'link3',
        }}
      />

      <OuiListGroupItem
        id="link4"
        onClick={() => {}}
        iconType="broom"
        label="OUI button link"
        extraAction={{
          color: 'subdued',
          onClick: link4Clicked,
          iconType: favorite4 === 'link4' ? 'starFilled' : 'starEmpty',
          iconSize: 's',
          'aria-label': 'Favorite link4',
          alwaysShow: favorite3 === 'link4',
          isDisabled: true,
        }}
      />

      <OuiListGroupItem
        id="link5"
        iconType="brush"
        isDisabled
        label="OUI button link"
        extraAction={{
          color: 'subdued',
          onClick: () => {},
          iconType: 'starEmpty',
          iconSize: 's',
          'aria-label': 'Favorite link4',
        }}
      />
    </OuiListGroup>
  );
};
