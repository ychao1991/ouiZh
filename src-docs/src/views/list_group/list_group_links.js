/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import { OuiListGroup } from '../../../../src/components';

const myContent = [
  {
    label: 'First link',
    href: '#/display/list-group',
    iconType: 'calendar',
    size: 's',
  },
  {
    label: 'This is an active link with very long label that truncates',
    href: '#/display/list-group',
    isActive: true,
    iconType: 'clock',
    size: 's',
  },
  {
    label: 'Third link is disabled',
    href: '#/display/list-group',
    isDisabled: true,
    iconType: 'compute',
    size: 's',
  },
  {
    label: 'Fourth link',
    href: '#/display/list-group',
    iconType: 'copyClipboard',
    size: 's',
  },
  {
    label: 'Fifth link',
    href: '#/display/list-group',
    iconType: 'crosshairs',
    size: 's',
  },
];

export default () => {
  return <OuiListGroup listItems={myContent} />;
};
