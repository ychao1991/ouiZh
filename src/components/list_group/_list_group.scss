/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/**
 * The List Group component provides neatly styled lists containing plain text
 * or links. The outer container can be bordered, with padding, or borderless
 * with links flush to the sides.
 */

.ouiListGroup {
  &.ouiListGroup-flush {
    padding: 0;
    border: none;
  }

  &.ouiListGroup-bordered {
    border-radius: $ouiBorderRadius;
    border: $ouiBorderThin;
  }
}

.ouiListGroup-maxWidthDefault {
  max-width: $ouiFormMaxWidth;
}

// Gutter Sizes
@each $gutterName, $gutterSize in $ouiListGroupGutterTypes {
  .ouiListGroup--#{$gutterName} {
    padding: $gutterSize;

    .ouiListGroupItem:not(:first-of-type) {
      margin-top: $gutterSize;
    }
  }
}
