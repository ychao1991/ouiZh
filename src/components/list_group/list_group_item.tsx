/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, {
  Fragment,
  HTMLAttributes,
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  ReactNode,
  ReactElement,
  MouseEventHandler,
  FunctionComponent,
} from 'react';
import classNames from 'classnames';

import { OuiButtonIcon, OuiButtonIconPropsForButton } from '../button';
import { OuiIcon, IconType, OuiIconProps } from '../icon';
import { OuiToolTip } from '../tool_tip';
import { useInnerText } from '../inner_text';
import { ExclusiveUnion, CommonProps } from '../common';

import { getSecureRelForTarget } from '../../services';
import { validateHref } from '../../services/security/href_validator';

type ItemSize = 'xs' | 's' | 'm' | 'l';
const sizeToClassNameMap: { [size in ItemSize]: string } = {
  xs: 'ouiListGroupItem--xSmall',
  s: 'ouiListGroupItem--small',
  m: 'ouiListGroupItem--medium',
  l: 'ouiListGroupItem--large',
};
export const SIZES = Object.keys(sizeToClassNameMap) as ItemSize[];

type Color = 'inherit' | 'primary' | 'text' | 'subdued' | 'ghost';
const colorToClassNameMap: { [color in Color]: string } = {
  inherit: '',
  primary: 'ouiListGroupItem--primary',
  text: 'ouiListGroupItem--text',
  subdued: 'ouiListGroupItem--subdued',
  ghost: 'ouiListGroupItem--ghost',
};
export const COLORS = Object.keys(colorToClassNameMap) as Color[];

export type OuiListGroupItemProps = CommonProps &
  Omit<
    ExclusiveUnion<
      ExclusiveUnion<
        ButtonHTMLAttributes<HTMLButtonElement>,
        Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'href'>
      >,
      HTMLAttributes<HTMLSpanElement>
    >,
    'onClick' | 'color' | 'target' | 'rel'
  > & {
    /**
     * Size of the label text
     */
    size?: ItemSize;
    /**
     * By default the item will inherit the color of its wrapper (button/link/span),
     * otherwise pass one of the acceptable options
     */
    color?: Color;

    /**
     * Content to be displayed in the list item
     */
    label: ReactNode;

    /**
     * Apply styles indicating an item is active
     */
    isActive?: boolean;

    /**
     * Apply styles indicating an item is disabled
     */
    isDisabled?: boolean;

    /**
     * Make the list item label a link.
     * While permitted, `href` and `onClick` should not be used together in most cases and may create problems.
     */
    href?: string;

    target?: string;

    rel?: string;

    /**
     * Adds `OuiIcon` of `OuiIcon.type`
     */
    iconType?: IconType;

    /**
     * Further extend the props applied to OuiIcon
     */
    iconProps?: Omit<OuiIconProps, 'type'>;

    /**
     * Custom node to pass as the icon. Cannot be used in conjunction
     * with `iconType` and `iconProps`.
     */
    icon?: ReactElement;

    /**
     * Display tooltip on list item
     */
    showToolTip?: boolean;

    /**
     * Adds an `OuiButtonIcon` to the right side of the item; `iconType` is required;
     * pass `alwaysShow` if you don't want the default behavior of only showing on hover
     */
    extraAction?: OuiButtonIconPropsForButton & {
      alwaysShow?: boolean;
    };

    /**
     * Make the list item label a button.
     * While permitted, `href` and `onClick` should not be used together in most cases and may create problems.
     */
    onClick?: MouseEventHandler<HTMLButtonElement>;

    /**
     * Allow link text to wrap
     */
    wrapText?: boolean;

    /**
     * Pass-through ref reference specifically for targeting
     * instances where the item content is rendered as a `button`
     */
    buttonRef?: React.Ref<HTMLButtonElement>;
  };

export const OuiListGroupItem: FunctionComponent<OuiListGroupItemProps> = ({
  label,
  isActive = false,
  isDisabled: _isDisabled = false,
  href,
  target,
  rel,
  className,
  iconType,
  icon,
  iconProps,
  extraAction,
  onClick,
  size = 'm',
  color = 'inherit',
  showToolTip = false,
  wrapText,
  buttonRef,
  ...rest
}) => {
  const isHrefValid = !href || validateHref(href);
  const isDisabled = _isDisabled || !isHrefValid;

  const classes = classNames(
    'ouiListGroupItem',
    sizeToClassNameMap[size],
    colorToClassNameMap[color],
    {
      'ouiListGroupItem-isActive': isActive,
      'ouiListGroupItem-isDisabled': isDisabled,
      'ouiListGroupItem-isClickable': href || onClick,
      'ouiListGroupItem-hasExtraAction': extraAction,
      'ouiListGroupItem--wrapText': wrapText,
    },
    className
  );

  let iconNode;

  if (iconType) {
    iconNode = (
      <OuiIcon
        color="inherit" // forces the icon to inherit its parent color
        {...iconProps}
        type={iconType}
        className={classNames('ouiListGroupItem__icon', iconProps?.className)}
      />
    );

    if (icon) {
      console.warn(
        'Both `iconType` and `icon` were passed to OuiListGroupItem but only one can exist. The `iconType` was used.'
      );
    }
  } else if (icon) {
    iconNode = React.cloneElement(icon, {
      className: classNames('ouiListGroupItem__icon', icon.props.className),
    });
  }

  let extraActionNode;

  if (extraAction) {
    const {
      iconType,
      alwaysShow,
      className,
      isDisabled: actionIsDisabled,
      ...rest
    } = extraAction;

    const extraActionClasses = classNames(
      'ouiListGroupItem__extraAction',
      {
        'ouiListGroupItem__extraAction-alwaysShow': alwaysShow,
      },
      className
    );

    extraActionNode = (
      <OuiButtonIcon
        className={extraActionClasses}
        iconType={iconType}
        {...rest}
        disabled={isDisabled || actionIsDisabled}
      />
    );
  }

  // Only add the label as the title attribute if it's possibly truncated
  // Also ensure the value of the title attribute is a string
  const [ref, innerText] = useInnerText();
  const shouldRenderTitle = !wrapText && !showToolTip;
  const labelContent = shouldRenderTitle ? (
    <span
      ref={ref}
      className="ouiListGroupItem__label"
      title={typeof label === 'string' ? label : innerText}>
      {label}
    </span>
  ) : (
    <span className="ouiListGroupItem__label">{label}</span>
  );

  // Handle the variety of interaction behavior
  let itemContent;

  const secureRel = getSecureRelForTarget({ href, rel, target });

  if (href && !isDisabled) {
    itemContent = (
      <a
        className="ouiListGroupItem__button"
        href={href}
        target={target}
        rel={secureRel}
        onClick={onClick as AnchorHTMLAttributes<HTMLAnchorElement>['onClick']}
        {...(rest as AnchorHTMLAttributes<HTMLAnchorElement>)}>
        {iconNode}
        {labelContent}
      </a>
    );
  } else if ((href && isDisabled) || onClick) {
    itemContent = (
      <button
        type="button"
        className="ouiListGroupItem__button"
        disabled={isDisabled}
        onClick={onClick}
        ref={buttonRef}
        {...(rest as ButtonHTMLAttributes<HTMLButtonElement>)}>
        {iconNode}
        {labelContent}
      </button>
    );
  } else {
    itemContent = (
      <span className="ouiListGroupItem__text" {...rest}>
        {iconNode}
        {labelContent}
      </span>
    );
  }

  if (showToolTip) {
    itemContent = (
      <li className={classes}>
        <OuiToolTip
          anchorClassName="ouiListGroupItem__tooltip"
          content={label}
          position="right"
          delay="long">
          {itemContent}
        </OuiToolTip>
      </li>
    );
  } else {
    itemContent = (
      <li className={classes}>
        {itemContent}
        {extraActionNode}
      </li>
    );
  }

  return <Fragment>{itemContent}</Fragment>;
};
