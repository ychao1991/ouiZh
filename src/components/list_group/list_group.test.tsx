/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { OuiListGroup, GUTTER_SIZES } from './list_group';
import { OuiListGroupItemProps } from './list_group_item';

const someListItems: OuiListGroupItemProps[] = [
  {
    label: 'Label with iconType',
    iconType: 'stop',
  },
  {
    label: 'Custom extra action',
    extraAction: {
      iconType: 'bell',
      alwaysShow: true,
      'aria-label': 'bell',
    },
  },
  {
    label: 'Button with onClick',
    onClick: (e) => {
      console.log('Visualize clicked', e);
    },
  },
  {
    label: 'Active link',
    isActive: true,
    href: '#',
  },
  {
    label: 'Link with href',
    href: '#',
  },
];

describe('OuiListGroup', () => {
  test('is rendered', () => {
    const component = render(<OuiListGroup {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  describe('listItems', () => {
    test('is rendered', () => {
      const component = render(<OuiListGroup listItems={someListItems} />);

      expect(component).toMatchSnapshot();
    });

    test('is rendered with color', () => {
      const component = render(
        <OuiListGroup color="primary" listItems={someListItems} />
      );

      expect(component).toMatchSnapshot();
    });

    test('is rendered with size', () => {
      const component = render(<OuiListGroup color="primary" size="xs" />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('props', () => {
    test('bordered is rendered', () => {
      const component = render(<OuiListGroup bordered />);

      expect(component).toMatchSnapshot();
    });

    test('flush is rendered', () => {
      const component = render(<OuiListGroup flush />);

      expect(component).toMatchSnapshot();
    });

    test('showToolTips is rendered', () => {
      const component = render(<OuiListGroup showToolTips />);

      expect(component).toMatchSnapshot();
    });

    test('wrapText is rendered', () => {
      const component = render(<OuiListGroup wrapText />);

      expect(component).toMatchSnapshot();
    });

    describe('gutter size', () => {
      GUTTER_SIZES.forEach((gutter) => {
        test(`${gutter} is rendered`, () => {
          const component = render(<OuiListGroup gutterSize={gutter} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('maxWidth', () => {
      test('as true is rendered', () => {
        const component = render(<OuiListGroup maxWidth={true} />);

        expect(component).toMatchSnapshot();
      });

      test('as a number is rendered', () => {
        const component = render(<OuiListGroup maxWidth={300} />);

        expect(component).toMatchSnapshot();
      });

      test('as a string is rendered', () => {
        const component = render(<OuiListGroup maxWidth="20em" />);

        expect(component).toMatchSnapshot();
      });
    });
  });
});
