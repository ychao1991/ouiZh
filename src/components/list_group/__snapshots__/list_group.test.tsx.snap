// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiListGroup is rendered 1`] = `
<ul
  aria-label="aria-label"
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`OuiListGroup listItems is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium"
  >
    <span
      class="ouiListGroupItem__text"
    >
      <span
        class="ouiListGroupItem__icon"
        color="inherit"
        data-ouiicon-type="stop"
      />
      <span
        class="ouiListGroupItem__label"
        title="Label with iconType"
      >
        Label with iconType
      </span>
    </span>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem-hasExtraAction"
  >
    <span
      class="ouiListGroupItem__text"
    >
      <span
        class="ouiListGroupItem__label"
        title="Custom extra action"
      >
        Custom extra action
      </span>
    </span>
    <button
      aria-label="bell"
      class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall ouiListGroupItem__extraAction ouiListGroupItem__extraAction-alwaysShow"
      type="button"
    >
      <span
        aria-hidden="true"
        class="ouiButtonIcon__icon"
        color="inherit"
        data-ouiicon-type="bell"
      />
    </button>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem-isClickable"
  >
    <button
      class="ouiListGroupItem__button"
      type="button"
    >
      <span
        class="ouiListGroupItem__label"
        title="Button with onClick"
      >
        Button with onClick
      </span>
    </button>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem-isActive ouiListGroupItem-isClickable"
  >
    <a
      class="ouiListGroupItem__button"
      href="#"
      rel="noreferrer"
    >
      <span
        class="ouiListGroupItem__label"
        title="Active link"
      >
        Active link
      </span>
    </a>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem-isClickable"
  >
    <a
      class="ouiListGroupItem__button"
      href="#"
      rel="noreferrer"
    >
      <span
        class="ouiListGroupItem__label"
        title="Link with href"
      >
        Link with href
      </span>
    </a>
  </li>
</ul>
`;

exports[`OuiListGroup listItems is rendered with color 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem--primary"
  >
    <span
      class="ouiListGroupItem__text"
    >
      <span
        class="ouiListGroupItem__icon"
        color="inherit"
        data-ouiicon-type="stop"
      />
      <span
        class="ouiListGroupItem__label"
        title="Label with iconType"
      >
        Label with iconType
      </span>
    </span>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem--primary ouiListGroupItem-hasExtraAction"
  >
    <span
      class="ouiListGroupItem__text"
    >
      <span
        class="ouiListGroupItem__label"
        title="Custom extra action"
      >
        Custom extra action
      </span>
    </span>
    <button
      aria-label="bell"
      class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall ouiListGroupItem__extraAction ouiListGroupItem__extraAction-alwaysShow"
      type="button"
    >
      <span
        aria-hidden="true"
        class="ouiButtonIcon__icon"
        color="inherit"
        data-ouiicon-type="bell"
      />
    </button>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem--primary ouiListGroupItem-isClickable"
  >
    <button
      class="ouiListGroupItem__button"
      type="button"
    >
      <span
        class="ouiListGroupItem__label"
        title="Button with onClick"
      >
        Button with onClick
      </span>
    </button>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem--primary ouiListGroupItem-isActive ouiListGroupItem-isClickable"
  >
    <a
      class="ouiListGroupItem__button"
      href="#"
      rel="noreferrer"
    >
      <span
        class="ouiListGroupItem__label"
        title="Active link"
      >
        Active link
      </span>
    </a>
  </li>
  <li
    class="ouiListGroupItem ouiListGroupItem--medium ouiListGroupItem--primary ouiListGroupItem-isClickable"
  >
    <a
      class="ouiListGroupItem__button"
      href="#"
      rel="noreferrer"
    >
      <span
        class="ouiListGroupItem__label"
        title="Link with href"
      >
        Link with href
      </span>
    </a>
  </li>
</ul>
`;

exports[`OuiListGroup listItems is rendered with size 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props bordered is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup-bordered ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props flush is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup-flush ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props gutter size m is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterMedium ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props gutter size none is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props gutter size s is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props maxWidth as a number is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall"
  style="max-width:300px"
/>
`;

exports[`OuiListGroup props maxWidth as a string is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall"
  style="max-width:20em"
/>
`;

exports[`OuiListGroup props maxWidth as true is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props showToolTips is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;

exports[`OuiListGroup props wrapText is rendered 1`] = `
<ul
  class="ouiListGroup ouiListGroup--gutterSmall ouiListGroup-maxWidthDefault"
/>
`;
